### ISSUE TITLE
Trigger system

### ISSUE THREAD

**Issue by DeputyLOL** [Mihara/RasterPropMonitor#357]
What am I missing? I went through the wiki and I couldn't find anything!
The triggers simply don't work for me right now. BUT... i'm gonna assume that its my fault.
This is a .cfg of my triggers:

RPM_TRIGGERED_EVENT
{
   eventName = autobrakespoiler
   variableName = CUSTOM_AUTOBRAKESPOILER
   range = 0.5,1.1
   triggerEvent = JSIFAR:SetSpoiler
   triggerState = true
   eventState = JSIFAR:GetSpoilerState
}

RPM_TRIGGERED_EVENT
{
   eventName = autobrake
   variableName = CUSTOM_AUTOBRAKE
   range = 0.5,1.1
   triggerEvent = brakes
   triggerState = false
   //eventState = not needed for triggerState = false
}

The variables are tested and they work and activate when expected.
Basically for both the variable is: gear down, master switch on(electricity present) and brake/spoiler arming switch on.

This is from the cockpit im using, I read on the wiki that I need to write these down:

 @PART[SXTmk3Cockpit52]:NEEDS[RasterPropMonitor]
 {
     @INTERNAL
     {
         @name = kossakintRPM
     }
     MODULE
     {
         name = RasterPropMonitorComputer
	 triggeredEvents = autobrakespoiler|autobrake	 
     }
 }

My problem is, the brakes and spoilers don't come on when the switch is pulled and the gear is down. Any ideas as whats wrong? 
(I did check to see if Norton did anything but it didn't.... this time.)
(And I am using the 0.23 dev version)

**Comment by MOARdV** [comment]
Check your logs - there should be some lines like "I know about event autobrakespolier" and "I know about event autobrake",  If they're not there, that means RPM didn't like the RPM_TRIGGERED_EVENT for some reason.  If they are there, the next question is whether the triggers were armed (the CUSTOM_ variables you're using have to be false at least once to arm the triggers).

You are setting brakes to 'false' according to that script, so I would expect them to be off, not on, when the trigger fires.  The FAR spoilers should fire, unless I did something to break the RPM FAR interface (check your log for a line like "JSIFAR: A supported version of FAR is present").

**Comment by DeputyLOL** [comment]
thanks for the reply, ill check the logs. The brakes are set to false for testing purposes. Then KSP crashed... 
Digging through the logs, I found this:
[LOG 22:32:15.291] RPMVesselComputer: Error adding triggered event autobrake: System.NullReferenceException: Object reference not set to an instance of an object
  at JSI.RPMVesselComputer+TriggeredEventTemplate..ctor (.ConfigNode node) [0x00000] in <filename unknown>:0 
  at JSI.RPMVesselComputer.Awake () [0x00000] in <filename unknown>:0 
[LOG 22:32:15.298] RPMVesselComputer: Error adding triggered event autobrake: System.NullReferenceException: Object reference not set to an instance of an object
  at JSI.RPMVesselComputer+TriggeredEventTemplate..ctor (.ConfigNode node) [0x00000] in <filename unknown>:0 
  at JSI.RPMVesselComputer.Awake () [0x00000] in <filename unknown>:0

**Comment by MOARdV** [comment]
Oops  It looks like I got a bit lazy testing some things.  I'll get a fix out today.

**Comment by MOARdV** [comment]
I double checked that your autobrake configuration works (I made up my own autobrake variable), so this should be okay now.  I've updated the dev build on DropBox - it also includes some logging for the event system (you'll see entries like "event autobrake armed" and "event autobrake triggered" if it works right)

**Comment by DeputyLOL** [comment]
ok thank you
